### PR TITLE
Refactor in order to simplify adding more tests

### PR DESCRIPTION
**Pull request by AlexanderLanin**
Besides looking better in any case ;-)
This is to support adding additional test cases for #219 

Review should be easier commit by commit. There are no changes besides moving from Facts to Theories.

Note: this is my first ever written code in C#, so feel free to be as strict as you wish.

### CODE REVIEW

File: src/shared/GitHub.Tests/GitHubHostProviderTests.cs [hunk 1]
```diff
@@ -97,73 +97,28 @@ public async Task GitHubHostProvider_GetSupportedAuthenticationModes(string uriS
         }
 
 
-
-        [Fact]
-        public async Task GitHubHostProvider_GetSupportedAuthenticationModes_NotDotCom_OldInstanceNoPassword_ReturnsNone()
+        [Theory]
+        [InlineData("https://example.com", null, "0.1", false, AuthenticationModes.None)]
+        [InlineData("https://example.com", null, "0.1", true, AuthenticationModes.Basic)]
+        [InlineData("https://example.com", null, "100.0", false, GitHubConstants.DotComAuthenticationModes)]
```
Comment (mjcheetham) [inline src/shared/GitHub.Tests/GitHubHostProviderTests.cs:103]: Can you replace the DotCom constant here with the explicit OAuth enumeration value please?
This is a test for the GitHub Enterprise mode discovery/logic and we don't want to make a link here to the DotCom modes constant.

```suggestion
        [InlineData("https://example.com", null, "100.0", false, AuthenticationModes.OAuth)]
```

**Review by mjcheetham — CHANGES_REQUESTED** [review]
Apart from the one comment, this looks good thanks! 👍

File: src/shared/GitHub.Tests/GitHubHostProviderTests.cs (hunk 1 above)
Comment (AlexanderLanin) [reply src/shared/GitHub.Tests/GitHubHostProviderTests.cs:103]: Sure. That's what was there before. This was intentionally without changing any logic.

**Review by mjcheetham — APPROVED** [review]
Thanks! Looks great to me.

**Pull request merged**